lib/adiwg/mdtranslator/readers/iso19115_2/modules/module_security_constraint.rb

```diff
@@ -0,0 +1,62 @@
+# frozen_string_literal: true
+
+require 'nokogiri'
+require 'adiwg/mdtranslator/internal/internal_metadata_obj'
+
+module ADIWG
+   module Mdtranslator
+      module Readers
+         module Iso191152
+            module SecurityConstraint
+               @@securityConstraintXPath = 'gmd:MD_SecurityConstraints'
+               @@classificationXPath = 'gmd:classification//gmd:MD_ClassificationCode'
+               @@userNoteXPath = 'gmd:userNote//gco:CharacterString'
+               @@classSysXPath = 'gmd:classificationSystem//gco:CharacterString'
+               @@handlingXPath = 'gmd:handlingDescription//gco:CharacterString'
+               @@type = 'security'
+               def self.unpack(xSecurityConstraint, hResponseObj)
+                  intMetadataClass = InternalMetadata.new
+                  hConstraint = intMetadataClass.newConstraint
+                  hSecConstraint = intMetadataClass.newSecurityConstraint
+
+                  hConstraint[:type] = @@type
+
+                  # xSecurityConstraint = xConstraint.xpath(@@securityConstraintXPath)[0]
+                  return nil if xSecurityConstraint.nil?
+
+                  # :classCode (required)
+                  # <gmd:MD_ClassificationCode codeList="http://mdtranslator.adiwg.org/api/codelists?format=xml#MD_ClassificationCode" codeListValue="classification" codeSpace="userCode"/>
```

Comment:
is this actually required? I think the comment copied down from the `19115_3` module.

Reply:
yes, `<xs:element name="classification" type="gmd:MD_ClassificationCode_PropertyType"/>` see [doc](https://www.isotc211.org/2005/gmd/constraints.xsd). looks for `MD_SecurityConstraints_Type`.

Reply:
you shouldn't paste an example of the element as evidence ( i.e. `# <gmd:MD_ClassificationCode codeList="http://mdtranslator.adiwg.org/api/codelists?format=xml#MD_ClassificationCode" codeListValue="classification" codeSpace="userCode"/>` ). you want to paste the exact element found in the corresponding .xsd file

Reply:
fixed with help from @rshewitt